docker/Dockerfile

```diff
@@ -0,0 +1,93 @@
+# Based on opencv-alpine by Juliano Petronetto (MIT Licence)
+
+FROM alpine:3.11
+ENV LANG=C.UTF-8
+
+RUN apk update && apk upgrade && apk --no-cache add \
+  binutils \
+  gcc \
+  g++ \
+  make \
+  cmake \
+  coreutils \
+  freetype-dev \
+  git \
+  libc-dev \
+  libffi-dev \
+  libpng-dev \
+  linux-headers \
+  lapack-dev \
+  openblas-dev \
+  unzip \
+  zlib-dev \
+  freeglut-dev
+
+RUN cd /opt && \
+  wget https://github.com/opencv/opencv/archive/3.2.0.zip && \
+  unzip 3.2.0.zip && rm 3.2.0.zip && \
+  cd opencv-3.2.0 && mkdir build && cd build && \
+  cmake -D CMAKE_BUILD_TYPE=RELEASE \
+    -D CMAKE_INSTALL_PREFIX=/usr/local \
+    -D INSTALL_C_EXAMPLES=OFF \
+    .. && \
+  make -j$(nproc) && \
+  make install && \
+  rm -rf /opt/opencv-3.2.0
+
+RUN cd /opt && \
+  wget https://downloads.sourceforge.net/project/arma/armadillo-9.850.1.tar.xz && \
+  tar xvf armadillo-9.850.1.tar.xz && \
+  rm armadillo-9.850.1.tar.xz && \
+  cd armadillo-9.850.1 && \
+  mkdir build && \
+  cd build && \
+  cmake .. -DCMAKE_INSTALL_PREFIX=/usr/local -DCMAKE_INSTALL_LIBDIR=/usr/local/lib && \
+  make -j$(nproc) && \
+  make install && \
+  rm -rf /opt/armadillo-9.850.1
+
+RUN cd /opt && \
+  wget https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.10/hdf5-1.10.6/src/hdf5-1.10.6.tar.gz && \
+  tar xvf hdf5-1.10.6.tar.gz && rm hdf5-1.10.6.tar.gz && \
+  cd hdf5-1.10.6 && \
+  mkdir build && \
+  cd build && \
+  cmake .. -DCMAKE_INSTALL_PREFIX=/usr/local && \
+  make -j$(nproc) && \
+  make install && \
+  cp /usr/local/lib/pkgconfig/hdf5-1.10.6.pc /usr/local/lib/pkgconfig/hdf5.pc && \
+  rm -rf /opt/hdf5-1.10.6
+
+RUN cd /opt && \
+  git clone https://github.com/open-source-parsers/jsoncpp.git && \
+  cd jsoncpp && \
+  git checkout 3beb37ea14aec1bdce1a6d542dc464d00f4a6cec && \
+  mkdir build && \
+  cd build && \
+  cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr/local -DCMAKE_INSTALL_LIBDIR=/usr/local/lib && \
+  make -j$(nproc) && \
+  make install && \
+  rm -rf /opt/jsoncpp
+
+#RUN apk add --no-cache libxrandr-dev libxinerama-dev libxcursor-dev && cd /opt && \
```

Comment:
@sebjameswml I had to comment this out. Otherwise morphologica fails to build:

```
[ 26%] Building CXX object tests/CMakeFiles/testDirichlet5.dir/testDirichlet5.cpp.oIn file included from /opt/morphologica/tests/../src/Process.h:23,
                 from /opt/morphologica/tests/../src/tools.h:28,
                 from /opt/morphologica/tests/testDirichlet5.cpp:10:
/usr/include/sys/poll.h:1:2: warning: #warning redirecting incorrect #include <sys/poll.h> to <poll.h> [-Wcpp]
    1 | #warning redirecting incorrect #include <sys/poll.h> to <poll.h>
      |  ^~~~~~~[ 27%] Linking CXX executable testboundary/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: libmorphologica.so: undefined reference to `glCreateBuffers'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: libmorphologica.so: undefined reference to `glCreateVertexArrays'
collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/testboundary.dir/build.make:110: src/testboundary] Error 1
make[1]: *** [CMakeFiles/Makefile2:133: src/CMakeFiles/testboundary.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 28%] Linking CXX executable testDirichlet5
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: ../src/libmorphologica.so: undefined reference to `glCreateBuffers'
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: ../src/libmorphologica.so: undefined reference to `glCreateVertexArrays'
collect2: error: ld returned 1 exit status
make[2]: *** [tests/CMakeFiles/testDirichlet5.dir/build.make:110: tests/testDirichlet5] Error 1
make[1]: *** [CMakeFiles/Makefile2:399: tests/CMakeFiles/testDirichlet5.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
```

It appears that morphologica could not link to the OpenGL library.

Reply:
I see those kinds of problems, and it can be associated with not having set the correct preprocessor definitions between including the GL headers.

Reply:
Oh, sorry, no, those are linker errors.